extensions/wikia/PageHeader/PageTitle.class.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+namespace PageHeader;
+
+class PageTitle {
+
+    /* @var string */
+    private $title;
+
+    /* @var \WikiaGlobalRegistry */
+    private $wg;
+
+    /* @var int */
+    private $namespace;
+
+    /**
+     * @param \WikiaApp $app
+     */
+    public function __construct($app) {
+        $this->wg = $app->wg;
+
+        $this->title = $app->getSkinTemplateObj()->data['title'];
+        $this->namespace = $this->wg->Title->getNamespace();
+    }
+
+    /**
+     * @return string
+     */
+    private function titleSpecial() {
+        return $this->title;
+    }
+
+    /**
+     * @return string
+     */
+    private function titleArticle() {
+        // remove namespaces prefix from title
+        $namespaces = [ NS_MEDIAWIKI, NS_TEMPLATE, NS_CATEGORY, NS_FILE ];
+
+        if (
+            in_array( $this->namespace,
+                array_merge( $namespaces, $this->wg->SuppressNamespacePrefix )
+            )
+        ) {
+            $title = $this->wg->Title->getText();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "$title" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1481)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)